### PR TITLE
[SofaCUDA] Add ConstantForceField, MultiMapping and SubsetMultiMapping

### PR DESCRIPTION
**Pull request by alxbilger**
1. Add CUDA support for ConstantForceField, MultiMapping and SubsetMultiMapping. The components are not accelerated by CUDA. Only compatibility with CUDA types is added.
2. Each section of the CMakeLists is sorted alphabetically.
3. An example scene is introduced for SubsetMultiMapping

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaSubsetMultiMapping.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <SofaMiscMapping/SubsetMultiMapping.inl>
+#include <sofa/core/ObjectFactory.h>
+#include "CudaTypes.h"
```
Comment (epernod) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaSubsetMultiMapping.cpp]:
```suggestion
#include <sofa/gpu/cuda/CudaTypes.h>
```

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaMultiMapping.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sofa/core/MultiMapping.inl>
+#include "CudaTypes.h"
```
Comment (epernod) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaMultiMapping.cpp]:
```suggestion
#include <sofa/gpu/cuda/CudaTypes.h>
```

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaConstantForceField.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <SofaBoundaryCondition/ConstantForceField.inl>
+#include <sofa/core/ObjectFactory.h>
+#include "CudaTypes.h"
```
Comment (epernod) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaConstantForceField.cpp]:
```suggestion
#include <sofa/gpu/cuda/CudaTypes.h>
```

**Review by epernod — COMMENTED** [review]
just minor changes

**Comment by epernod** [comment]
Could you check #2556 
If you agree with the new directories, maybe you could put the new scene inside:
SofaCUDA/scenes/examples/  or SofaCUDA/scenes/demos/

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaConstantForceField.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+{
+    using namespace sofa::component::forcefield;
+
+    // Register in the Factory
+    int ConstantForceFieldCudaClass = core::RegisterObject("Constant forces applied to given degrees of freedom")
+        .add< ConstantForceField<CudaVec3Types> >()
+        .add< ConstantForceField<CudaVec2Types> >()
+        .add< ConstantForceField<CudaVec1Types> >()
+        .add< ConstantForceField<CudaVec6Types> >()
+        // .add< ConstantForceField<CudaRigid3Types> >() //Rigid3Types is a valid template, but CudaRigid3Types is not
```
Comment (damienmarchal) [inline applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaConstantForceField.cpp]: Remove commented code.

**Comment by alxbilger** [comment]
@epernod the new scene is now in SofaCUDA/scenes/examples/

**Comment by alxbilger** [comment]
Actually, I accidentally deleted the scene instead of moving it. I'll put it back in another PR

**Pull request merged**